medusa/logger/__init__.py

```diff
@@ -219,34 +219,23 @@ def blocks_r(filename, size=64 * 1024, reset_offset=True, encoding='utf-8'):
         size (int|None): The block size.
             If int, the file is yielded in blocks of the specified size.
             If None, the file is yielded at once.
-        reset_offset (bool): Reset the file offset.
-            If True, starts reading from the end of the file.
-            Otherwise, starts reading from where the file current position is.
-        encoding (str|None): The encoding for correct block size computation.
-            If `str`, must be a valid string encoding.
-            If None, the default encoding is used.
 
     Yields:
-        block (str): The data within the blocks.
+        block (bytes): The data within the blocks.
 
     """
-    with io.open(filename, 'r', encoding=encoding) as file_obj:
-        offset = 0
-        if reset_offset:
-            file_size = remaining_size = file_obj.seek(0, os.SEEK_END)
-        else:
-            file_size = remaining_size = file_obj.tell()
-        rounding = 0
+    with io.open(filename, 'rb') as file_obj:
+        remaining_size = file_obj.seek(0, os.SEEK_END)
         while remaining_size > 0:
-            offset = min(file_size, offset + size)
-            file_obj.seek(file_size - offset)
-            block = file_obj.read(min(remaining_size, size))
-            remaining_size -= size
-            yield block[:len(block) + rounding] if rounding else block
+            block_size = min(remaining_size, size)
+            file_obj.seek(remaining_size - block_size)
+            block = file_obj.read(block_size)
+            remaining_size -= block_size
+            yield block
 
 
-def reverse_readlines(filename, skip_empty=True, append_newline=False, block_size=512 * 1024,
-                      reset_offset=True, encoding='utf-8'):
+def reverse_readlines(filename, skip_empty=True, append_newline=False,
+                      block_size=128 * 1024, encoding='utf-8'):
```

Comment:
Do you really think that there's a reason to lower the block size?
512KB to 128KB?

Reply:
Considering that we only show [max 1000 lines in the log viewer](https://github.com/pymedusa/Medusa/blob/develop/themes-default/slim/src/components/logs.vue#L153) , which is roughly 1000 (this can vary ofc). I don't see why we should keep such a high block size. Also, [this](http://rabexc.org/posts/io-performance-in-python) seems to suggest that 128k is the default `MMAP_THRESHOLD` (that doesn't necessarily mean that it is more performant tho).

Reply:
Okay then, I don't know enough about it but sure.
I guess it's the same thing as the `shutil` monkeypatch, where if the block size is larger than X, it doesn't have any noticeable improvement to performance.